subiquity/models/subiquity.py

```diff
@@ -203,10 +203,18 @@ def _cloud_init_files(self):
         # for subsequent boots.
         # (mwhudson does not entirely know what the above means!)
         userdata = '#cloud-config\n' + yaml.dump(self._cloud_init_config())
-        metadata = yaml.dump({'instance-id': str(uuid.uuid4())})
+        metadata = {'instance-id': str(uuid.uuid4())}
+        config = {
+            'datasource_list': ["None"],
+            'datasource': {
+                "None": {
+                    'userdata_raw': userdata,
+                    'metadata': metadata,
+                    },
+                },
+            }
         files = [
-            ('var/lib/cloud/seed/nocloud-net/meta-data', metadata, 0o644),
-            ('var/lib/cloud/seed/nocloud-net/user-data', userdata, 0o600),
+            ('etc/cloud/cloud.cfg.d/installer.cfg', yaml.dump(config), 0o600),
```

Comment:
I would change this to 99-subiquity.cfg;  using number > 90 will ensure this clobers the built-in 90-dpkg.cfg and I think including subiquity is useful to know what program wrote this (also consider a # comment in the yaml with the signature as well.

Reply:
That makes sense, I didn't realise that that's what 90_dpkg.cfg had in it. I'm a bit confused as to how this branch helps with the cloud.cfg.d snippets being in the order they are, but it does...

Reply:
This is likely due to luck/accident.  ds-identify does not have a full yaml parser, but rather searches for keys while globbing through /etc/cloud/cloud.cfg /etc/cloud/cloud.cfg.d/*.cfg;

Reply:
Oh yay. I've updated the filename.